pkg/sanity/sanity.go

```diff
@@ -449,3 +459,86 @@ func PseudoUUID() string {
 func UniqueString(prefix string) string {
 	return prefix + uniqueSuffix
 }
+
+// Return codes for CheckPath
+type PathKind string
+
+const (
+	PathIsFile     PathKind = "file"
+	PathIsDir      PathKind = "directory"
+	PathIsNotFound PathKind = "not_found"
+	PathIsOther    PathKind = "other"
+)
+
+func (pk PathKind) validate() error {
+	var err error = nil
+	switch pk {
+	case PathIsFile:
+	case PathIsDir:
+	case PathIsNotFound:
+	case PathIsOther:
+	default:
+		err = fmt.Errorf("invalid PathType: %s", pk)
+	}
+	return err
```

Comment:
This seems non-idiomatic. A shorter version is:
```
func (pk PathKind) validate() error {
	switch pk {
	case PathIsFile, PathIsDir, PathIsNotFound, PathIsOther:
              return nil
         default:
		return fmt.Errorf("invalid PathType: %s", pk)
	}
}
```

Reply:
I also find it odd to first cast a string and then call validate. It works of course, but a `func IsPathKind (in string) (PathKind, error)` makes the intent clearer.

Reply:
Makes sense, done. Appreciate your help with this.